src/main/java/org/orekit/propagation/events/EclipseDetector.java

```diff
@@ -178,7 +221,15 @@ public boolean getTotalEclipse() {
      */
     public double g(final SpacecraftState s) {
         final Vector3D pted  = occulted.getPVCoordinates(s.getDate(), occulting.getBodyFrame()).getPosition();
-        final Vector3D psat  = s.getPVCoordinates(occulting.getBodyFrame()).getPosition();
+        Vector3D psat = null;
+        if (subSatellitePoint) {
+    		Frame inertialFrame = FramesFactory.getEME2000();
```

Comment:
The inertial frame should be extracted from SpacecraftState rather than hard-coded to EME2000.
Otherwise, the s.getPVCoordinates().getPosition() in the following line will be inconsistent.

Reply:
I fixed it.